src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -706,36 +706,42 @@ public static ServerInfo GetServerVersion(SqlConnectionStringBuilder csb, string
                 azureAccountToken: azureAccountToken);
 
             return serverInfo;
-        }
-
-        /// <summary>
-        /// Gets the server host information from sys.dm_os_host_info view
-        /// </summary>
-        /// <param name="connection">The connection</param>
+        }
+
+        /// <summary>
+        /// Gets the server host information from sys.dm_os_host_info view
+        /// </summary>
+        /// <param name="connection">The connection</param>
         public static ServerHostInfo GetServerHostInfo(IDbConnection connection)
         {
-            // SQL Server 2016 and below does not provide sys.dm_os_host_info
+            var hostInfo = new ServerHostInfo();
+            // SQL Server 2016 and earlier versions does not provide sys.dm_os_host_info and we know the host OS can only be Windows.
             if (!Version.TryParse(ReadServerVersion(connection), out var hostVersion) || hostVersion.Major <= 13)
             {
-                return new ServerHostInfo
-                {
-                    Platform = "Windows"
-                };
+                ExecuteReader(
+                    connection,
+                    SqlConnectionHelperScripts.GetHostWindowsVersion,
```

Comment:
in this case "!Version.TryParse(ReadServerVersion(connection), out var hostVersion)" if GetHostWindowsVersion runs it will cause failure - right? Should we try catch it?

Reply:
good catch, I am catching the error and ignore it, that falls back to the previous behavior of only setting the platform to  Windows.